salt/modules/junos.py

```diff
@@ -939,29 +945,31 @@ def install_config(path=None, **kwargs):
             finally:
                 salt.utils.files.safe_rm(template_cached_path)
 
-            config_diff = cu.diff()
-            if config_diff is None:
-                ret['message'] = 'Configuration already applied!'
-                ret['out'] = True
-                return ret
+            if db_mode != 'dynamic':
+                config_diff = cu.diff()
+                if config_diff is None:
+                    ret['message'] = 'Configuration already applied!'
+                    ret['out'] = True
+                    return ret
 
             commit_params = {}
             if 'confirm' in op:
                 commit_params['confirm'] = op['confirm']
             if 'comment' in op:
                 commit_params['comment'] = op['comment']
 
-            try:
-                check = cu.commit_check()
-            except Exception as exception:
-                ret['message'] = \
-                    'Commit check threw the following exception: "{0}"'\
-                    .format(exception)
+            if db_mode != 'dynamic':
+                try:
+                    check = cu.commit_check()
+                except Exception as exception:
+                    ret['message'] = \
+                        'Commit check threw the following exception: "{0}"'\
+                        .format(exception)
 
-                ret['out'] = False
-                return ret
+                    ret['out'] = False
+                    return ret
 
-            if check and not test:
+            if db_mode == 'dynamic' or check and not test:
```

Comment:
can you just use default value for check and then we wont need this line change

Reply:
`commit_check` returns `True` if commit_check succeeds, else raises an exception.

Link: https://github.com/Juniper/py-junos-eznc/blob/dabef6dc29af8017d5eb8ae4be70daa7a42b77d6/lib/jnpr/junos/utils/config.py#L175